### PR TITLE
boards: factor out shared code for arduino-due and udoo

### PR DESCRIPTION
**Pull request by haukepetersen**
EDIT
~~based on #8058~~

The `udoo` and the `arduino-due` are basically the same boards, sharing everything down to the last pin. This is now reflected in the board structure.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
Will review once #8058 is merged. Please rebase then.

**Comment by haukepetersen** [comment]
will do, thx

**Comment by haukepetersen** [comment]
dependencies were merged, so this PR is ready for review.

**Comment by PeterKietzmann** [comment]
I know, I know :-). There are a couple of candidates to review now... Will try to look at this one later

File: boards/common/arduino_due/include/w5100_params.h [hunk 1]
```diff
@@ -34,10 +36,10 @@ extern "C" {
 #define W5100_PARAM_SPI_CLK     (SPI_CLK_5MHZ)
 #endif
 #ifndef W5100_PARAM_CS
-#define W5100_PARAM_CS          (GPIO_PIN(2, 29))
+#define W5100_PARAM_CS          (ARDUINO_PIN_10)
```
Comment (PeterKietzmann) [inline boards/common/arduino_due/include/w5100_params.h]: Nearly missed that change above :-)

Comment (haukepetersen) [reply boards/common/arduino_due/include/w5100_params.h]: what do you mean with missed? Do you disagree with this change?

Comment (PeterKietzmann) [reply boards/common/arduino_due/include/w5100_params.h]: No forget it, everything 's fine with it

**Comment by haukepetersen** [comment]
I see your point for the underscore in the common name -> fixed this.

About the name `arduio-due` for the common folder: the is actually the best option. something like `arduino32, -cortex, or -m3` or similar won't work here, as the other 32-bit arduinos have very differing options to most extends. The this here is, that the `udoo` people actually replicated the `due` to the detail, so `arduino-due` as shared folder makes sense here.

In follow-up work there might be however aspects of this PR that are candidates for even more generalization, as to take parts of the `common/arduino-due` code and share it with the mkr's. But another day.

**Pull request merged**